src/cloudflare.rs

```diff
@@ -297,11 +317,13 @@ impl<'a> CloudflareClient<'a> {
             content: addr.to_string(),
         };
 
-        let response: CloudflareResponse<CloudflareDnsRecord> = self
+        let mut request_builder: reqwest::RequestBuilder = self
             .client
-            .patch(&url)
-            .header("X-Auth-Email", self.email.clone())
-            .header("X-Auth-Key", self.key.clone())
+            .patch(&url);
+
+            request_builder = apply_auth_to_request(request_builder, self.email.clone(), self.key.clone(), self.token.clone());
+                
+            let response: CloudflareResponse<Vec<CloudflareDnsRecord>> = request_builder
```

Comment:
I believe:

```
CloudflareResponse<Vec<CloudflareDnsRecord>>
```

should be

```
CloudflareResponse<CloudflareDnsRecord>
```

I receive an error trying to interpret cloudflare's response (the update is still effective, it's just that an error is printed on updates).

Reply:
Oops. Yeah you're right. I've pushed a fixed version.

Reply:
Excellent, run `cargo fmt` in the repo to format the code and this PR will look to be all set to be merged. I tested it out and it worked well 🎉

Reply:
Great. I've done that and pushed a new commit since the format changes probably spanned across multiple commits and I'm not sure if you just intend to squash them all upon merge anyway. Let me know if you'd like me to take an alternative approach.

Reply:
You're good. My plan is to cut a patch release with the current master, merge this PR, touch things up a bit, and then cut another release in a week or so. Thanks again